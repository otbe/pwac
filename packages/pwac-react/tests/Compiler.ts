import { execSync } from 'npm-run';
import { join, resolve } from 'path';
import { snapshot } from 'snapshot-dir';

describe('Compiler (react)', () => {
  const cwd = resolve(__dirname, '..', '..', 'example-react');
  process.env.PWAC_CONFIG_TEST = 'test_config';

  it(`should build the example with config from package.json`, async () => {
    try {
      await execSync('pwac build', {
        cwd
      });
    } catch (e) {
      expect(e).toBeUndefined();
    }

    const result = await snapshot(join(cwd, 'dist'));

    expect(result).toMatchSnapshot();
  });

  [
    'complete',
    'disabled',
    'manifest',
    'prerender-disabled',
    'preload-lazy-chunks',
    'image-optimization-disabled'
  ].forEach(c => {
    it(`should build the example with config ${c}.json`, async () => {
      try {
        await execSync('pwac build', {
          cwd,
          env: {
            ...process.env,
            PWAC_CONFIG: resolve(__dirname, 'configs', `${c}.json`)
          }
        });
      } catch (e) {
        expect(e).toBeUndefined();
      }

      const result = await snapshot(join(cwd, 'dist'));

      expect(result).toMatchSnapshot();
    });
  });
});
