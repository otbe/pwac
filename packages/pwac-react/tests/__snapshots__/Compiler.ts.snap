// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compiler (react) should build the example with config complete.json 1`] = `
Object {
  "1.cee7ea9abce823baf990.css": "16e4925d2f0de443907834b33b46394b",
  "1.cee7ea9abce823baf990.js": "fb98bc73e5bf09820dfc71e9a694db43",
  "1.cee7ea9abce823baf990.js.map": "cb7e806928f6052db8e9c9ff6e395790",
  "2.bf18163718a8126bae33.js": "8f9d8e8c9db493b414bed4793cc4de65",
  "2.bf18163718a8126bae33.js.map": "22d805fbd9fa24f53b067e640c8c5a30",
  "240683d2dd7c5a158ee48c70703a2a0a-1200.jpg": "240683d2dd7c5a158ee48c70703a2a0a",
  "5c917367a9adfba3f1e904ed920580be-600.jpg": "5c917367a9adfba3f1e904ed920580be",
  "8f42a999073e8fe5caf83406f0e1d690-1920.jpg": "8f42a999073e8fe5caf83406f0e1d690",
  "c29e3724f0ae14c777b61c09ee20e73b-300.jpg": "c29e3724f0ae14c777b61c09ee20e73b",
  "index.html": "0f446e018ee22a88872eeb89e63c66d1",
  "main.eea9587ec3d0c93b66a0.css": "7861def0e3928a6ff8b473840ce73a4e",
  "main.eea9587ec3d0c93b66a0.js": "d724b655ad18aab73defccf5efd94448",
  "main.eea9587ec3d0c93b66a0.js.map": "f554599c2df8fb267f3335591429d7ec",
  "manifest.5331242021cd9f5a0938.js": "47eec6f49052bf28a3bb60df3b78f274",
  "manifest.5331242021cd9f5a0938.js.map": "10b1aa3ca188e7d40c2fb3ea903710f1",
  "sw.js": "edc8c0e7e1f0a3fd84e33f8ebd655cfe",
  "vendors.9641a7591b585946bdac.css": "a184b7f85fa905337909488ff5eee01b",
  "vendors.9641a7591b585946bdac.js": "8383cf11c0f30b449e4b2ccd58499e6f",
  "vendors.9641a7591b585946bdac.js.map": "44b01f556fa384004ca3c9dfd7bc3698",
}
`;

exports[`Compiler (react) should build the example with config disabled.json 1`] = `
Object {
  "1.cee7ea9abce823baf990.css": "16e4925d2f0de443907834b33b46394b",
  "1.cee7ea9abce823baf990.js": "fb98bc73e5bf09820dfc71e9a694db43",
  "1.cee7ea9abce823baf990.js.map": "cb7e806928f6052db8e9c9ff6e395790",
  "2.bf18163718a8126bae33.js": "8f9d8e8c9db493b414bed4793cc4de65",
  "2.bf18163718a8126bae33.js.map": "22d805fbd9fa24f53b067e640c8c5a30",
  "b02330cbe0a6272c907bdb4e12b97b7c.jpg": "be5bdc38e858972b37353622a1079182",
  "index.html": "7db80f82342f4472f0c95c21896f2581",
  "main.0713ce1d917e53acccc8.css": "7861def0e3928a6ff8b473840ce73a4e",
  "main.0713ce1d917e53acccc8.js": "428a643879109dc49b309178544932d0",
  "main.0713ce1d917e53acccc8.js.map": "c63cbe0b545e699104a7779ff83758a7",
  "manifest.5331242021cd9f5a0938.js": "47eec6f49052bf28a3bb60df3b78f274",
  "manifest.5331242021cd9f5a0938.js.map": "10b1aa3ca188e7d40c2fb3ea903710f1",
  "sw.js": "8a18b1fda49d6f569a9689e871a11603",
  "vendors.9641a7591b585946bdac.css": "a184b7f85fa905337909488ff5eee01b",
  "vendors.9641a7591b585946bdac.js": "8383cf11c0f30b449e4b2ccd58499e6f",
  "vendors.9641a7591b585946bdac.js.map": "44b01f556fa384004ca3c9dfd7bc3698",
}
`;

exports[`Compiler (react) should build the example with config from package.json 1`] = `
Object {
  "1.cee7ea9abce823baf990.css": "16e4925d2f0de443907834b33b46394b",
  "1.cee7ea9abce823baf990.js": "fb98bc73e5bf09820dfc71e9a694db43",
  "1.cee7ea9abce823baf990.js.map": "cb7e806928f6052db8e9c9ff6e395790",
  "2.bf18163718a8126bae33.js": "8f9d8e8c9db493b414bed4793cc4de65",
  "2.bf18163718a8126bae33.js.map": "22d805fbd9fa24f53b067e640c8c5a30",
  "240683d2dd7c5a158ee48c70703a2a0a-1200.jpg": "240683d2dd7c5a158ee48c70703a2a0a",
  "5c917367a9adfba3f1e904ed920580be-600.jpg": "5c917367a9adfba3f1e904ed920580be",
  "8f42a999073e8fe5caf83406f0e1d690-1920.jpg": "8f42a999073e8fe5caf83406f0e1d690",
  "bar/index.html": "d9c085b4f3437595a5e60e352c2332f6",
  "c29e3724f0ae14c777b61c09ee20e73b-300.jpg": "c29e3724f0ae14c777b61c09ee20e73b",
  "icon_128x128.deb0a118869f6d69fcc5e9852ad32d81.png": "deb0a118869f6d69fcc5e9852ad32d81",
  "icon_192x192.f5d0f0f2ed01fca7cb69e70f6af11529.png": "f5d0f0f2ed01fca7cb69e70f6af11529",
  "icon_256x256.4661d370dd248c9f9d48e78c8a07776d.png": "4661d370dd248c9f9d48e78c8a07776d",
  "icon_96x96.1c0003f895382900e1a8ce6fa0a1b672.png": "1c0003f895382900e1a8ce6fa0a1b672",
  "index.html": "210229a3e9734ce6faf11b9c65b6666c",
  "main.eea9587ec3d0c93b66a0.css": "7861def0e3928a6ff8b473840ce73a4e",
  "main.eea9587ec3d0c93b66a0.js": "d724b655ad18aab73defccf5efd94448",
  "main.eea9587ec3d0c93b66a0.js.map": "f554599c2df8fb267f3335591429d7ec",
  "manifest.5331242021cd9f5a0938.js": "47eec6f49052bf28a3bb60df3b78f274",
  "manifest.5331242021cd9f5a0938.js.map": "10b1aa3ca188e7d40c2fb3ea903710f1",
  "manifest.bc9f52edf87c052e960e4516d360c190.json": "bc9f52edf87c052e960e4516d360c190",
  "sw.js": "901d3798d05fc176fe71df9686e6cef2",
  "vendors.9641a7591b585946bdac.css": "a184b7f85fa905337909488ff5eee01b",
  "vendors.9641a7591b585946bdac.js": "8383cf11c0f30b449e4b2ccd58499e6f",
  "vendors.9641a7591b585946bdac.js.map": "44b01f556fa384004ca3c9dfd7bc3698",
}
`;

exports[`Compiler (react) should build the example with config image-optimization-disabled.json 1`] = `
Object {
  "1.cee7ea9abce823baf990.css": "16e4925d2f0de443907834b33b46394b",
  "1.cee7ea9abce823baf990.js": "fb98bc73e5bf09820dfc71e9a694db43",
  "1.cee7ea9abce823baf990.js.map": "cb7e806928f6052db8e9c9ff6e395790",
  "2.bf18163718a8126bae33.js": "8f9d8e8c9db493b414bed4793cc4de65",
  "2.bf18163718a8126bae33.js.map": "22d805fbd9fa24f53b067e640c8c5a30",
  "b02330cbe0a6272c907bdb4e12b97b7c.jpg": "be5bdc38e858972b37353622a1079182",
  "index.html": "5dc82ba871493eb46944553442594fa1",
  "main.0713ce1d917e53acccc8.css": "7861def0e3928a6ff8b473840ce73a4e",
  "main.0713ce1d917e53acccc8.js": "428a643879109dc49b309178544932d0",
  "main.0713ce1d917e53acccc8.js.map": "c63cbe0b545e699104a7779ff83758a7",
  "manifest.5331242021cd9f5a0938.js": "47eec6f49052bf28a3bb60df3b78f274",
  "manifest.5331242021cd9f5a0938.js.map": "10b1aa3ca188e7d40c2fb3ea903710f1",
  "sw.js": "8a18b1fda49d6f569a9689e871a11603",
  "vendors.9641a7591b585946bdac.css": "a184b7f85fa905337909488ff5eee01b",
  "vendors.9641a7591b585946bdac.js": "8383cf11c0f30b449e4b2ccd58499e6f",
  "vendors.9641a7591b585946bdac.js.map": "44b01f556fa384004ca3c9dfd7bc3698",
}
`;

exports[`Compiler (react) should build the example with config manifest.json 1`] = `
Object {
  "1.cee7ea9abce823baf990.css": "16e4925d2f0de443907834b33b46394b",
  "1.cee7ea9abce823baf990.js": "fb98bc73e5bf09820dfc71e9a694db43",
  "1.cee7ea9abce823baf990.js.map": "cb7e806928f6052db8e9c9ff6e395790",
  "2.bf18163718a8126bae33.js": "8f9d8e8c9db493b414bed4793cc4de65",
  "2.bf18163718a8126bae33.js.map": "22d805fbd9fa24f53b067e640c8c5a30",
  "b02330cbe0a6272c907bdb4e12b97b7c.jpg": "be5bdc38e858972b37353622a1079182",
  "icon_128x128.deb0a118869f6d69fcc5e9852ad32d81.png": "deb0a118869f6d69fcc5e9852ad32d81",
  "icon_192x192.f5d0f0f2ed01fca7cb69e70f6af11529.png": "f5d0f0f2ed01fca7cb69e70f6af11529",
  "icon_256x256.4661d370dd248c9f9d48e78c8a07776d.png": "4661d370dd248c9f9d48e78c8a07776d",
  "icon_96x96.1c0003f895382900e1a8ce6fa0a1b672.png": "1c0003f895382900e1a8ce6fa0a1b672",
  "index.html": "f2e514f3859319ab10e6876cbfca6ad9",
  "main.0713ce1d917e53acccc8.css": "7861def0e3928a6ff8b473840ce73a4e",
  "main.0713ce1d917e53acccc8.js": "428a643879109dc49b309178544932d0",
  "main.0713ce1d917e53acccc8.js.map": "c63cbe0b545e699104a7779ff83758a7",
  "manifest.5331242021cd9f5a0938.js": "47eec6f49052bf28a3bb60df3b78f274",
  "manifest.5331242021cd9f5a0938.js.map": "10b1aa3ca188e7d40c2fb3ea903710f1",
  "manifest.bc9f52edf87c052e960e4516d360c190.json": "bc9f52edf87c052e960e4516d360c190",
  "sw.js": "f21ecb13983eaabb0ed76d70cafe3b17",
  "vendors.9641a7591b585946bdac.css": "a184b7f85fa905337909488ff5eee01b",
  "vendors.9641a7591b585946bdac.js": "8383cf11c0f30b449e4b2ccd58499e6f",
  "vendors.9641a7591b585946bdac.js.map": "44b01f556fa384004ca3c9dfd7bc3698",
}
`;

exports[`Compiler (react) should build the example with config preload-lazy-chunks.json 1`] = `
Object {
  "1.cee7ea9abce823baf990.css": "16e4925d2f0de443907834b33b46394b",
  "1.cee7ea9abce823baf990.js": "fb98bc73e5bf09820dfc71e9a694db43",
  "1.cee7ea9abce823baf990.js.map": "cb7e806928f6052db8e9c9ff6e395790",
  "2.bf18163718a8126bae33.js": "8f9d8e8c9db493b414bed4793cc4de65",
  "2.bf18163718a8126bae33.js.map": "22d805fbd9fa24f53b067e640c8c5a30",
  "240683d2dd7c5a158ee48c70703a2a0a-1200.jpg": "240683d2dd7c5a158ee48c70703a2a0a",
  "5c917367a9adfba3f1e904ed920580be-600.jpg": "5c917367a9adfba3f1e904ed920580be",
  "8f42a999073e8fe5caf83406f0e1d690-1920.jpg": "8f42a999073e8fe5caf83406f0e1d690",
  "c29e3724f0ae14c777b61c09ee20e73b-300.jpg": "c29e3724f0ae14c777b61c09ee20e73b",
  "index.html": "0f446e018ee22a88872eeb89e63c66d1",
  "main.eea9587ec3d0c93b66a0.css": "7861def0e3928a6ff8b473840ce73a4e",
  "main.eea9587ec3d0c93b66a0.js": "d724b655ad18aab73defccf5efd94448",
  "main.eea9587ec3d0c93b66a0.js.map": "f554599c2df8fb267f3335591429d7ec",
  "manifest.5331242021cd9f5a0938.js": "47eec6f49052bf28a3bb60df3b78f274",
  "manifest.5331242021cd9f5a0938.js.map": "10b1aa3ca188e7d40c2fb3ea903710f1",
  "sw.js": "39ad85198d89c7782ca2d0ed80538a66",
  "vendors.9641a7591b585946bdac.css": "a184b7f85fa905337909488ff5eee01b",
  "vendors.9641a7591b585946bdac.js": "8383cf11c0f30b449e4b2ccd58499e6f",
  "vendors.9641a7591b585946bdac.js.map": "44b01f556fa384004ca3c9dfd7bc3698",
}
`;

exports[`Compiler (react) should build the example with config prerender-disabled.json 1`] = `
Object {
  "1.cee7ea9abce823baf990.css": "16e4925d2f0de443907834b33b46394b",
  "1.cee7ea9abce823baf990.js": "fb98bc73e5bf09820dfc71e9a694db43",
  "1.cee7ea9abce823baf990.js.map": "cb7e806928f6052db8e9c9ff6e395790",
  "2.bf18163718a8126bae33.js": "8f9d8e8c9db493b414bed4793cc4de65",
  "2.bf18163718a8126bae33.js.map": "22d805fbd9fa24f53b067e640c8c5a30",
  "240683d2dd7c5a158ee48c70703a2a0a-1200.jpg": "240683d2dd7c5a158ee48c70703a2a0a",
  "5c917367a9adfba3f1e904ed920580be-600.jpg": "5c917367a9adfba3f1e904ed920580be",
  "8f42a999073e8fe5caf83406f0e1d690-1920.jpg": "8f42a999073e8fe5caf83406f0e1d690",
  "c29e3724f0ae14c777b61c09ee20e73b-300.jpg": "c29e3724f0ae14c777b61c09ee20e73b",
  "index.html": "3def8b69a39732590fcdccd7be0debd3",
  "main.eea9587ec3d0c93b66a0.css": "7861def0e3928a6ff8b473840ce73a4e",
  "main.eea9587ec3d0c93b66a0.js": "d724b655ad18aab73defccf5efd94448",
  "main.eea9587ec3d0c93b66a0.js.map": "f554599c2df8fb267f3335591429d7ec",
  "manifest.5331242021cd9f5a0938.js": "47eec6f49052bf28a3bb60df3b78f274",
  "manifest.5331242021cd9f5a0938.js.map": "10b1aa3ca188e7d40c2fb3ea903710f1",
  "sw.js": "edc8c0e7e1f0a3fd84e33f8ebd655cfe",
  "vendors.9641a7591b585946bdac.css": "a184b7f85fa905337909488ff5eee01b",
  "vendors.9641a7591b585946bdac.js": "8383cf11c0f30b449e4b2ccd58499e6f",
  "vendors.9641a7591b585946bdac.js.map": "44b01f556fa384004ca3c9dfd7bc3698",
}
`;
